src/plugins/discover/public/application/apps/main/components/top_nav/get_top_nav_links.ts

```diff
@@ -149,12 +157,66 @@ export const getTopNavLinks = ({
     },
   };
 
+  const dataVisualizer = {
```

Comment:
As Discover is OSS and the Index Data Visualizer is not, should this be in Discover?
This is the same problem I faced when adding the File Data Visualizer component to Home's Add Data page, and it was decided that the File Data Visualizer plugin should register itself with Home when starting up, rather than have Home know anything about File Data Visualizer plugin.

Reply:
Related to this, if I create a space with no access to the ML feature privilege, the link to the Data visualizer is still added to the top nav, but when clicking on it a page opens with a 404 error.

```
{"statusCode":404,"error":"Not Found","message":"Not Found"}
```

With the current setup where the data visualizer page is inside the ML plugin, we should only add the link if ML is enabled in the space.

Reply:
You're right. It would be better to have a mechanism for plugins like data visualizer to register links to Discover instead 👍 I'll also revisit the permissions check.